main/src/main/scala/sbt/Script.scala

```diff
@@ -32,7 +42,7 @@ object Script {
       val append = Load.transformSettings(Load.projectScope(currentRef), currentRef.build, rootProject, scriptSettings ++ embeddedSettings)
 
       val newStructure = Load.reapply(session.original ++ append, structure)
-      val arguments = state.remainingCommands.drop(1)
+      val arguments = state.remainingCommands.drop(1).map(arg => "\"" + arg + "\"")
```

Comment:
Where are you running the second example from? If it's something like Bash, `"` gets interpreted by that, right?